src/reformat.c

```diff
@@ -1008,17 +1077,21 @@ avifResult avifImageYUVToRGB(const avifImage * image, avifRGBImage * rgb)
         return AVIF_RESULT_REFORMAT_FAILED;
     }
 
-    avifBool convertedWithLibYUV = AVIF_FALSE;
-    avifResult libyuvResult = avifImageYUVToRGBLibYUV(image, rgb);
-    if (libyuvResult == AVIF_RESULT_OK) {
-        convertedWithLibYUV = AVIF_TRUE;
-    } else {
-        if (libyuvResult != AVIF_RESULT_NOT_IMPLEMENTED) {
-            return libyuvResult;
-        }
+    // libyuv uses integer math which is less precise.
+    // Avoid libyuv if user is requesting best quality.
+    avifResult convertResult = AVIF_RESULT_NOT_IMPLEMENTED;
+    if (rgb->chromaUpsampling != AVIF_CHROMA_UPSAMPLING_BEST_QUALITY) {
+        convertResult = avifImageYUVToRGBLibYUV(image, rgb);
     }
 
-    if (avifRGBFormatHasAlpha(rgb->format) && !rgb->ignoreAlpha) {
+    if (convertResult == AVIF_RESULT_OK) {
+        return AVIF_RESULT_OK;
+    } else if (convertResult != AVIF_RESULT_NOT_IMPLEMENTED && convertResult != AVIF_RESULT_NEED_POST_PROCESS) {
+        return convertResult;
+    }
+
+    // XRGB is considered as RGB which need premultiplied pixel, so we still need alpha if we need to do premultiply.
```

Comment:
Nit: Remove "do" before "premultiply" because we can use "premultiply" as a verb.